tasks/dataverse-glassfish.yaml

```diff
@@ -1,93 +1,81 @@
+---
+# dataverse/tasks/dataverse-glassfish.yaml
+
 - name: download and unzip glassfish4
   get_url:
     url: http://dlc-cdn.sun.com/glassfish/4.1/release/glassfish-4.1.zip
     dest: /tmp
-  tags: glassfish
+    # checksum: sha1:704a90899ec5e3b5007d310b13a6001575827293 # requires ansible ≥2.0
 
 - name: unzip glassfish
-  shell: "unzip -d /tmp /tmp/glassfish-4.1.zip"
-  tags: glassfish
+  unarchive:
```

Comment:
`unarchive` is idempotent, can be run repeatedly, and correctly registers changed/ok; `shell: unzip …` cannot. In general, we should shy away from using shell where an Ansible module exists to handle the problem. I understand this might be an issue for developers running older (pre-2.0) versions of Ansible. In those cases, I would highly recommend upgrading your Ansible version — it's not a hard update, and there are a number of improvements in 2.x.